src/FormBuilder/resources/js/form-builder/src/components/canvas/FormSettings.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useMemo} from 'react';
+import FormSettingsContainer from '@givewp/form-builder/components/canvas/FormSettingsContainer';
+import FormDonationConfirmationSettingsGroup from '@givewp/form-builder/settings/group-donation-confirmation';
+import FormGeneralSettingsGroup from '@givewp/form-builder/settings/group-general';
+import {__} from '@wordpress/i18n';
+import EmailGeneralSettings from '@givewp/form-builder/settings/group-email-settings/general';
+import getEmailSettings from '@givewp/form-builder/settings/group-email-settings';
+
+export default function FormSettings() {
+    const routes: Route[] = useMemo(() => {
+        const additionalSettings = wp.hooks.applyFilters('givewp_form_builder_settings_additional_routes', []);
+
+        return [
+            {
+                name: __('General', 'give'),
+                path: 'general',
```

Comment:
Soo, this is something i'd like to discuss..

_What's the point of a route having a path?_

If this was an actual browser router, it would be more obvious as the path would correspond to the url - but if our routes are simply an array of objects that gets shown and hidden - why even does it even make sense to add the overhead of having a unique path name?  It seems in our case the path is just a unique identity.

The alternative i'm thinking of would be to generate the route ID based on its position in the array of routes as integers. Child routes could work the same way with some kind of simple naming/syntax convention.

Reply:
I could go either way. Having a unique id not based on position is helpful, as position based identifiers can run into strange issues if things fire in a different order (for whatever reason). This feels like very low overhead and would allow us to add hash routing in the future if we so chose.

Reply:
@JasonTheAdams what do you mean by "if things fire in a different order" in this context?  Is there something specifically your concerned about?

The overhead i'm initially concerned about is this gives the responsibility to the route to determine its unique path and will need to compete (validation) with other existing path names. While that doesn't seem like a big deal right now, i'm looking towards the future, with the addition of many more settings & add-ons, the path will become an important piece to manage with these settings.  

If there is a clear benefit of that approach and/or we _do_ intentionally plan on having a hash router i'm all ears.  Otherwise, the alternative approach for generated ids could offer less friction and maintenance over time.

Reply:
I'm thinking in terms of things like the `key` prop in React, which must always be unique, wherein using the index is problematic if the order or amount of items change. Since add-ons can affect the routes, we just want to make sure that isn't a problem.

The reason I'm most inclined to think we may add a hash or something in the future is so that settings are linkable. Say, for example, we have a screen elsewhere that says "To change x go to this setting", and links you to the settings page in the form. This is also useful for support to be able to send customers direct links as to where they can change a setting.

Reply:
hmm how would support be able to send someone a link to a form setting without knowing the form id?  I suppose its possible but you would also need to explain to the customer about form ids.  I'm curious if we're doing that now in any capacity, like giving someone a link and telling them to concat their form ID in the URL. 

In general, hash routing is just not something i've seen implemented inside a Gutenberg page yet.

I'm not completely against having the path names, i'm just trying to weigh the options & consequences.

It seems the main advantage of route defined path name is _flexibility_ in that we could potentially add some kind of url routing in the future and maybe more flexibility in terms of programmatic positioning as well.  However, thse are all theoretical at the moment.

The downsides are mainly the overhead of maintaining path names for routes, the need to validate those path names so there are no duplicates & extra documentation for adding additional routes.

Reply:
The path is being used here as an unique id for each menu item so we can identify what's the current active route, rendering its content and marking the menu item as active as well. The path also helps to identify at what level is that selected route. For instance: `email-settings/general`. Splitting that string on every `/`, we can detect that the `general` is in the level 2. As the menu works with pages like in a slider, I can know that the General item is the one in the second page.

I thought to use uuid to dynamically generate that but I was afraid to have numbers changing on every rendering besides having an unnecessary computational work for each route.

Reply:
@pauloiankoski yes, I understand what they are used for and i'm not even debating use a unique identifier. I'm just weighing the options of dynamically generating the paths like this:

```
routes.map((route, index) => ({
	id: `givewp-setting-${index}`
	element: route.element,
	childRoutes: route.childRoutes.map((route, index) => ({
		id: `givewp-setting-${index}/givewp-setting-${index}`
		element: route.element,
	}))
}))
```

Reply:
Refactored route elements with `{settings, setSettings}` props: 64fa35e7b865cca1787fd252fa9182fa94b302d4